### PR TITLE
feat(drawer): describe using design tokens (FE-4738)

### PR DESCRIPTION
**Pull request by Parsium**
### Proposed behaviour

Update `Drawer` so its appearance is only described using Design Tokens, to give designers more control over its styling.

- Remove `baseTheme` references in `*.style.js` files
- Set default background of sidebar container to `colors.utility.major.040`, and border right to `colorsUtilityMajor075`
- Set background colour, padding and box shadow of sidebar footer using the respective design tokens
- Amend control button, drawer header to use design tokens.
- Amend tests to verify new styling.
- Reorganise a couple of the tests for readability.
- Update storybook stories to work with new styling 

### Current behaviour

`Drawer`'s appearance is described using the styled-components `theme` property.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

- `Drawer` internally uses `Box`, which has custom styling for the scrollbar that's not part of Design System. @harpalsingh will double check with the designers whether this existing styling is adequate. (original PR which added this scrollbar styling can be found here: #3478)
- Typography-based design tokens have **not** been added - this will be addressed in a later ticket (FE-4923).

### Testing instructions

Double check component's appearance in storybook matches with the official designs and their documentation

### CODE REVIEW

**Review by DipperTheDan — DISMISSED** [review]

**Review by jhajduk-sage — DISMISSED** [review]

File: src/__internal__/sticky-footer/sticky-footer.style.js [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
 `;
 
 const StyledStickyFooter = styled.div`
-  ${({ sticky, theme }) => css`
+  padding: var(--spacing200) var(--spacing400);
+  box-sizing: border-box;
+  width: 100%;
+  ${({ sticky }) => css`
     ${sticky &&
     css`
```
Comment (grabkowski) [inline src/__internal__/sticky-footer/sticky-footer.style.js:14]:
```suggestion
  $({ sticky }) => sticky && css`
```

File: src/__internal__/sticky-footer/sticky-footer.style.js [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
 
 const FormButtonAnimation = keyframes`
   0%   { transform: translateY(50px); }
   100% { transform: translateY(0px); }
 `;
 
 const StyledStickyFooter = styled.div`
-  ${({ sticky, theme }) => css`
+  padding: var(--spacing200) var(--spacing400);
+  box-sizing: border-box;
```
Comment (grabkowski) [inline src/__internal__/sticky-footer/sticky-footer.style.js:10]: looks like these css properties were not part of this component, and component will change visually after they are added. They should be added as separate commit, e.g.: `feat(drawer): add padding to sticky footer`

**Review by DipperTheDan — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by jhajduk-sage — DISMISSED** [review]

**Comment by harpalsingh** [comment]
The Drawer is missing the border right as you can see from the design it should have one but its now missing:

**should be:**
Screenshot 2022-02-08 at 08 54 18

**PR shows:** 
Screenshot 2022-02-08 at 08 54 54

**Review by DipperTheDan — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by jhajduk-sage — DISMISSED** [review]

**Comment by harpalsingh** [comment]
The border right color is incorrect here, should be token "Utility Major 075"

**Review by DipperTheDan — DISMISSED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by jhajduk-sage — APPROVED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]
